uxarray/grid/grid.py

```diff
@@ -7,87 +7,111 @@
-    islatlon : bool, optional
-        Specify if the grid is lat/lon based
-    isconcave: bool, optional
-        Specify if this grid has concave elements (internal checks for this are possible)
-    use_dual: bool, optional
-        Specify whether to use the primal (use_dual=False) or dual (use_dual=True) mesh if the file type is mpas
-
-    Raises
-    ------
-        RuntimeError
-            If specified file not found or recognized
+    grid_ds : xr.Dataset
+        ``xarray.Dataset`` encoded in the UGRID conventions
+
+    source_grid_spec : str, default="UGRID"
+        Original unstructured grid format (i.e. UGRID, MPAS, etc.)
+
+    source_dims_dict : dict, default={}
+        Mapping of dimensions from the source dataset to their UGRID equivalent (i.e. {nCell : nMesh2_face})
 
     Examples
     ----------
 
     >>> import uxarray as ux
+    >>> grid_path = "/path/to/grid.nc"
+    >>> data_path = "/path/to/data.nc"
 
     1. Open a grid file with `uxarray.open_grid()`:
 
-    >>> uxgrid = ux.open_grid("filename.g")
+    >>> uxgrid = ux.open_grid(grid_path)
 
     2. Open an unstructured grid dataset file with
-    `uxarray.open_dataset()`, then access `Grid` info:
+    `uxarray.open_dataset()`, then access the ``Grid``.:
 
-    >>> uxds = ux.open_dataset("filename.g")
+    >>> uxds = ux.open_dataset(grid_path, data_path)
+    >>> uxds.uxgrid
     """
 
-    def __init__(self, input_obj, **kwargs):
-        # initialize internal variable names
-        self.__init_grid_var_names__()
+    def __init__(self,
+                 grid_ds: xr.Dataset,
+                 source_grid_spec: Optional[str] = None,
+                 source_dims_dict: Optional[dict] = {}):
 
-        # initialize face_area variable
-        self._face_areas = None
+        # check if inputted dataset is a minimum representable 2D UGRID unstructured grid
+        if not _validate_minimum_ugrid(grid_ds):
+            raise ValueError(
+                "Use of Grid constructor requires grid_ds to follow the internal unstructured grid definition, "
+                "including variable and dimension names. This grid_ds does not satisfy those requirements. If you are "
+                "not sure about how to do that, using ux.open_grid() or ux.from_dataset() is suggested."
+            )  # TODO: elaborate once we have a formal definition
+
+        # grid spec not provided, check if grid_ds is a minimum representable UGRID dataset
+        if source_grid_spec is None:
+            warn(
+                "Attempting to construct a Grid without passing in source_grid_spec. Use of Grid constructor"
+                "is only advised if grid_ds is following the internal unstructured grid definition, including"
+                "variable and dimension names. Using ux.open_grid() or ux.from_dataset() is suggested.",
+                Warning)
+
+            # TODO: more checks for validate grid (lat/lon coords, etc)
+            source_grid_spec = "UGRID"
```

Comment:
```suggestion
```

Reply:
If `source_grid_spec` is given `None`, fine, we shouldn't do anything. Giving only a warning should be enough (you may want to rephrase the warning text one more time to emphasize `sourcegrid_spec` is for diagnostics and reporting purposes only, hence a warning not an error).